x/ibc_query/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,107 @@
+		msgQueryResult  types.MsgSubmitCrossChainQueryResult
+		IBCQueryKeeper  ibcquerykeeper.Keeper
+		ctx             sdk.Context
+		mocks           testkeeper.MockedKeepers
+    )
+
+    testCases := []struct {
+        msg             string
+		expPass         bool
+        setUp           func()
+		queryMsg        types.QueryCrossChainQueryResult
+		expQueryResult  types.QueryCrossChainQueryResultResponse
+    }{
+        {
+            "success: query valid query id", true,
+			func () {
+				IBCQueryKeeper, ctx, _, mocks = testkeeper.GetIBCQueryKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
+				msgSubmitQuery = testkeeper.GetMsgSubmitCrossChainQuery(ctx, "test/path", senderAddr.String())
+				msgQueryResult = testkeeper.GetMsgSubmitCrossChainQueryResult(ctx, "query-0", senderAddr.String())
+			},
+			types.QueryCrossChainQueryResult{
+				Id: "query-0",
+			},
+			types.QueryCrossChainQueryResultResponse{
+				Id:       "query-0",
+				Result:   types.QueryResult_QUERY_RESULT_SUCCESS,
+				Data:     []byte("test result data"),
+			},
+        },
+		{
+            "success: query valid query id", true,
+			func () {
+				IBCQueryKeeper, ctx, _, mocks = testkeeper.GetIBCQueryKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
+				msgSubmitQuery = testkeeper.GetMsgSubmitCrossChainQuery(ctx, "test/path", senderAddr.String())
+				msgQueryResult = testkeeper.GetMsgSubmitCrossChainQueryResult(ctx, "query-0", senderAddr.String())
+				gomock.InOrder(testkeeper.GetMocksForSubmitCrossChainQuery(ctx, &mocks, "query-0", msgSubmitQuery.Sender)...)
+				IBCQueryKeeper.SubmitCrossChainQuery(ctx, &msgSubmitQuery)   
+				IBCQueryKeeper.SubmitCrossChainQueryResult(ctx, &msgQueryResult)
+
+				msgQueryResult = testkeeper.GetMsgSubmitCrossChainQueryResult(ctx, "query-1", senderAddr.String())
+			},
+			types.QueryCrossChainQueryResult{
+				Id: "query-1",
+			},
+			types.QueryCrossChainQueryResultResponse{
+				Id:       "query-1",
+				Result:   types.QueryResult_QUERY_RESULT_SUCCESS,
+				Data:     []byte("test result data"),
+			},
+        },
+		{
+            "fail: query invalid query id", false,
+			func () {
+				IBCQueryKeeper, ctx, _, mocks = testkeeper.GetIBCQueryKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
+				msgSubmitQuery = testkeeper.GetMsgSubmitCrossChainQuery(ctx, "test/path", senderAddr.String())
+				msgQueryResult = testkeeper.GetMsgSubmitCrossChainQueryResult(ctx, "query-0", senderAddr.String())
+			},
+			types.QueryCrossChainQueryResult{
+				Id: "query-1",
+			},
+			types.QueryCrossChainQueryResultResponse{},
+        },
+    }
+
+    for _, tc := range testCases {
+		tc.setUp()
+
+		// Run SubmitCrossChainQuery
+		gomock.InOrder(testkeeper.GetMocksForSubmitCrossChainQuery(ctx, &mocks, testkeeper.GetQueryId(IBCQueryKeeper, ctx), msgSubmitQuery.Sender)...)
+		IBCQueryKeeper.SubmitCrossChainQuery(ctx, &msgSubmitQuery)
+		IBCQueryKeeper.SubmitCrossChainQueryResult(ctx, &msgQueryResult)   
```

Comment:
why don't you validate return value of these function?
This testcode should verify "SubmitCrossChainQuery" and "SubmitCrossChainQueryResult" working

Reply:
Because "SubmitCrossChainQuery" and "SubmitCrossChainQueryResult" working is verified in msg_server_test.go, i thought i didn't need it. This file is intended to verify CrossChainQueryResult function working